### PR TITLE
Fixed initial focus to ignore hidden elements

### PR DESCRIPTION
**Pull request by blackbaud-conorwright**
Resolves: #11 
Docs: N/A

### CODE REVIEW

File: src/app/public/modules/modal/fixtures/modal-with-focus-content.fixture.html [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <sky-modal>
   <sky-modal-header>Test</sky-modal-header>
   <sky-modal-content>
-    <button>Content</button>
+    <button style="visibility: hidden">Hidden</button>
+    <button style="display: none">No display</button>
```
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/modal/fixtures/modal-with-focus-content.fixture.html]: These are great. Should we have an element with 0 size too? The adapter function checks for bounds along with `visibility` and `display`.

Comment (blackbaud-conorwright) [reply src/app/public/modules/modal/fixtures/modal-with-focus-content.fixture.html]: Oh yeah, it didn't have a test for that. This is actually to check if it is displayed as well (has a bounds and a visual offset in the viewport) I'll add one of the other display states it covers. Nice catch 👍

Comment (blackbaud-conorwright) [reply src/app/public/modules/modal/fixtures/modal-with-focus-content.fixture.html]: Alright, so IE and Edge do not seem to support the only other option I came across `display: contents`, so covering this in the test will fail in IE and Edge. Unfortunately I had to take that out

File: src/app/public/modules/modal/modal-component-adapter.service.ts [hunk 2]
```diff
@@ -113,8 +113,13 @@ export class SkyModalComponentAdapterService {
   }
 
   private isVisible(element: HTMLElement) {
-    return !!(element.offsetWidth ||
+    const style = window.getComputedStyle(element);
+    const isHidden = style.display === 'none' || style.visibility === 'hidden';
+    const hasBounds = !!(
```
[the remark this answers is not in the document]
Comment (blackbaud-conorwright) [reply src/app/public/modules/modal/modal-component-adapter.service.ts]: done
